### PR TITLE
Counting sort in C Language

### LINKED ISSUE #745: Counting sort Algo in python and c
@jainaman224  I want to work on this issue.

### PR DESCRIPTION
**Pull request by shadow989**
Contribution in GSsoc19 Counting Sort in C
fixed dynamic array input for sorting

Fixes #745 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Sample Input / Output.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

#### Changes proposed in this pull request:

- Dynamic array input added
- proper input output added

#### Languages Used:
- c language
- 

#### Files Added:
- counting_sort.c
-

### CODE REVIEW

**Comment by jainaman224** [comment]
@shadow989 Please complete this PR template.

File: counting_sort.c [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    }
+    printf("The sorted array is:\n");
+    for(i=1;i<=n;i++)
+    {
+        printf("%d\t",c[i]);
+    }
+    //to hold the result in turboc++ it takes a single charater and program ends
+    getch();
+}
+
```
Comment (jainaman224) [inline counting_sort.c]: Add sample input output.

File: counting_sort.c [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        c[b[a[j]]] = a[j];
+        b[a[j]] = b[a[j]] - 1;
+    }
+    printf("The sorted array is:\n");
+    for(i=1;i<=n;i++)
+    {
+        printf("%d\t",c[i]);
+    }
+    //to hold the result in turboc++ it takes a single charater and program ends
+    getch();
```
Comment (jainaman224) [inline counting_sort.c]: Remove this. Please write code for standard gcc compiler.

File: counting_sort.c [hunk 3]
```diff
@@ -0,0 +1,60 @@
+//Implementation of counting sort in c language
+
+#include <conio.h>
+#include <stdio.h>
```
Comment (jainaman224) [inline counting_sort.c]: Conio is not supported in standard gcc compiler.

File: counting_sort.c [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+int main() 
+{
+	// initializing the variables of use
```
Comment (jainaman224) [inline counting_sort.c]: Fix indentation.

File: counting_sort.c [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+    int *a ,*c,i,max=0,j,n;
+
```
Comment (jainaman224) [inline counting_sort.c]: Use descriptive variable names.

**Comment by shadow989** [comment]
@jainaman224  review once more i have completed everything.

**Comment by jainaman224** [comment]
Which issue you are solving?

**Comment by shadow989** [comment]
@jainaman224 #866 issue

**Comment by jainaman224** [comment]
It is PR number. You must have claimed issue before working on it?

**Comment by shadow989** [comment]
@jainaman224  #745

**Comment by jainaman224** [comment]
Sorry, but code already exists. I am closing this PR. Will give suitable marks. Sorry for the inconvenience caused.

**Pull request closed without merging**